src/beam.ts

```diff
@@ -11,11 +11,10 @@ import { Fraction } from './fraction';
 import { Tuplet } from './tuplet';
 import { Stem } from './stem';
 import { Note } from './note';
-import { StemmableNote } from './stemmablenote';
 import { Voice } from './voice';
 import { RenderContext } from './types/common';
 
-function calculateStemDirection(notes: StemmableNote[]) {
+function calculateStemDirection(notes: Note[]) {
```

Comment:
These should remain `StemmableNote` -- beams don't make sense unless there are stems.

Reply:
Sure but:
- the code is getting the contents as `Note` from `Voice` 
https://github.com/0xfe/vexflow/blob/ec03fb3cf194c48b67c95f6798b0e9bec5491c2e/src/beam.ts#L134
- the code is expecting the notes not to be `StemmableNote`:
https://github.com/0xfe/vexflow/blob/ec03fb3cf194c48b67c95f6798b0e9bec5491c2e/src/beam.ts#L682-L683

Reply:
So for `applyAndGetBeams`:

```typescript
static applyAndGetBeams(voice: Voice, stem_direction?: number, groups?: Fraction[]): Beam[] {
    const stemmables = (voice.getTickables() as Array<StemmableNote>);
    return Beam.generateBeams(stemmables, {
      groups,
      stem_direction,
    });
  }
```

The second part is effectively a run-time type check for `StemmableNote` and throws an exception if there's no stem -- if it was all typescript, you can just remove that check.

Reply:
This `as` approach bypasses the TypeScripts checks. The current approach with the functions in `Note` calling RERR make sure that we do not make an error. I think that this is better, is not it?

Reply:
Not really -- getTickables() returns `Note` which is type checked, and `Note` is explicitly typecast to `StemmableNote` -- so the window of vulnerability is pretty small. This is not bypassing checks entirely like using `any`.

You can close the vulnerability with a typeguard: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

This approach is way cleaner than absorbing `StemmableNote` methods into `Note`, and then accepting `Note` in `Voice`.

Reply:
No doubt that this is not as bad as `any`!
Are you proposing to use `as` and then `instanceof` to check if the assumption is good? Have a look at the fiddle:

https://jsfiddle.net/zorjyg8L/

Reply:
Well not really needed. If there is an error it will be catched in the accepted changes in Note where we ar both not really fans. :)

Reply:
done.